### ISSUE TITLE
xUnit1051 code fix chooses incorrect overload for CancellationToken

### ISSUE THREAD

**Issue by SapiensAnatis** [xunit/xunit#3068]
I ran into this while upgrading to v3 and with EF Core's FindAsync method inside my tests, which exposes a `params object[] keyValues` overload and an `object[] keyValues, CancellationToken cancellationToken` overload.

Given the following code:

```cs
public class UnitTest1
{
    [Fact]
    public async Task Test1()
    {
        await M(1);
    }
    
    private async Task M(params int[] numbers)
    {
        
    }

    private async Task M(int[] numbers, CancellationToken cancellationToken = default)
    {
        
    }
}
```

I get a diagnostic for xUnit1051 on the invocation of `M` in UnitTest1. But if I apply the code fix, I get 

```cs
[Fact]
public async Task Test1()
{
    await M(1, TestContext.Current.CancellationToken);
}
```

which is a compile error. The correct code is:

```cs
[Fact]
public async Task Test1()
{
    await M([1], TestContext.Current.CancellationToken);
}
```

There's an argument here that the EF Core API is not the best (ReSharper does warn me about the second method being hidden). In fact, it's a bit worse in that case since `params object[]` accepts the CancellationToken but just leads to a runtime failure by trying to treat it as a primary key.

That being said, I didn't see any existing issues with this so I thought I'd raise it all the same - maybe it's an easy fix.

**Comment by bradwilson** [comment]
I'm tempted, given the `params object[]` problem, to say that we should just say that those two methods are not compatible and don't trigger xUnit1051 at all.

**Comment by SapiensAnatis** [comment]
That sounds reasonable to me. Although, I wonder what [CA2016 - Forward the CancellationToken parameter to methods that take one](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2016) does in this case?

I might be able to work on this by the way, I have a little bit of experience with analyzers.

**Comment by SapiensAnatis** [comment]
The code fix for CA2016 in this instance will create a new array for you when it switches to the other overload:

```diff
public class C
{
    public async Task M(CancellationToken cancellationToken = default)
    {
-      await M2(1, 2, 3);
+      await M2(new int[] { 1, 2, 3 }, cancellationToken: cancellationToken);
    }

    public async Task M2(params int[] values)
    {
    }

    public async Task M2(int[] values, CancellationToken cancellationToken = default)
    {
    }
}
```

So it's perhaps not unreasonable to attempt making xUnit1051 do the same. However, the complexity of this may have increased a bit with the new support in C# 13 for non-array `params`, which even CA2016 doesn't appear to handle properly.

**Comment by SapiensAnatis** [comment]
I've gone with copying CA2016 and attempting to create an array, since it wasn't that hard to do in the end (assuming my solution works). Let me know if you'd prefer to go the route of not firing this diagnostic in the  first place, though.

**Comment by bradwilson** [comment]
This sounds good, I'll take a look at the PR.

**Comment by bradwilson** [comment]
Available in `1.18.0-pre.6` https://xunit.net/docs/using-ci-builds
